src/frequenz/sdk/actor/_runner.py

```diff
@@ -0,0 +1,41 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Utility functions to run and synchronize the execution of actors."""
+
+
+import asyncio
+import logging
+from typing import Any
+
+from ._decorator import BaseActor
+
+log = logging.getLogger(__name__)
+
+
+async def run(*actors: Any) -> None:
+    """Await the completion of all actors.
+
+    Args:
+        actors: the actors to be awaited.
+
+    Raises:
+        AssertionError: if any of the actors is not an instance of BaseActor.
+    """
+    actor_tasks = []
+    # Check that each actor is an instance of BaseActor at runtime,
+    # due to the indirection created by the actor decorator.
+    for actor in actors:
+        assert isinstance(actor, BaseActor), f"{actor} is not an instance of BaseActor"
+        actor_tasks.append(asyncio.create_task(actor.join()))
+
+    done, pending = await asyncio.wait(actor_tasks, return_when=asyncio.ALL_COMPLETED)
```

Comment:
Doesn't `ALL_COMPLETED` guarantee that all task will be `done`? If so, I guess logging `pending` doesn't make a lot of sense, but since it is a debug log I'm fine with keeping it too. But if it is not guaranteed that all will be done, you need to put this in a loop that runs until all actors really stopped.

Reply:
`ALL_COMPLETED` guarantees all tasks are finished or are cancelled. Indeed, `pending` does not make sense in this case. I was initially using `pending` when `return_when` was set to `FIRST_EXCEPTION`.

The actor decorator handles the actor restarts on exceptions (or actor termination) within a loop so I don't think we need to re-do it here at the moment.

I'll remove the pending variable and related logging. Though `asyncio.wait()` returns a tuple but I guess I can use `done, _ = await asyncio.wait(...)` to avoid a linting error of a non-used variable.

Reply:
Removed pending, but I still kept the debugging messages for `done`

Reply:
Using `FIRST_COMPLETED` and looping might also be a good idea, so you get more *real-time* logging when actors finish running, and not just all the logging at the end. But we can leave this for a follow-up PR.

Reply:
it makes sense to me to add the loop and use `FIRST_COMPLETED`, though I'd keep the inline comment that `currently the actor decorator manages the life-cycle of the actor tasks` to avoid misunderstandings (I guess this might change on follow-up PRs).